jina/executors/encoders/audio/fairseq.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from .. import BaseAudioEncoder
+from ..frameworks import BaseTorchEncoder
+from ...decorators import batching, as_ndarray
+
+
+class Wav2VecSpeechEncoder(BaseTorchEncoder, BaseAudioEncoder):
+    """
+    :class:`Wav2VecSpeechEncoder` Wav2Vec-based speech encoder presented by Facebook https://github.com/pytorch/fairseq
+    that encodes an audio signal from a `Batch x Signal Length` into a `Batch * Frames x Features` ndarray.
+    """
+
+    def __init__(self, model_path: str, *args, **kwargs):
+        """
+        Wav2Vec produces a representation for each time step at a rate of 100 Hz.
+
+        :param model_path: the path of the pretrained model. The pretrained model can be downloaded at
+            https://dl.fbaipublicfiles.com/fairseq/wav2vec/wav2vec_large.pt
+        """
+        super().__init__(*args, **kwargs)
+        self.model_path = model_path
+
+    def post_init(self):
+        import torch
+        from fairseq.models.wav2vec import Wav2VecModel
+        cp = torch.load(self.model_path, map_location=torch.device('cpu'))
+        self.model = Wav2VecModel.build_model(cp['args'], task=None)
+        self.model.load_state_dict(cp['model'])
+        self.model.eval()
```

Comment:
It would be great if the codes can be run on both CPUs and GPUs. Please refer to the codes at [torchvision.py#L52](https://github.com/jina-ai/jina/blob/master/jina/executors/encoders/image/torchvision.py#L52)
```
self.to_device(self.model)
```